### DISCUSSION TITLE
Change before_event, after_event signature to be more pythonic

### DISCUSSION THREAD

**Discussion by Vitalium** [BeanieODM/beanie#354] [Feature Request]
Hello!

My proposal is to change signature of functions `{before,after}_event` from:
`def before_event(event_types: Union[List[EventTypes], EventTypes]):`
to:
`def before_event(*event_types: EventTypes):`

`@before_event([Insert, Update])` vs `@before_event(Insert, Update)`

It looks more elegant and "pythonic" in both source and client code.

**Comment by roman-right** [comment]
Hey! This sounds reasonable. I'll try to add this today. Thank you

**Comment by roman-right** [comment]
Implemented in `1.11.12`

**Comment by Vitalium** [comment]
👍 Great job! Thank you for taking into account.

By the way, typing hint for starred positional arguments can be as simple as the type of tuple entries (`*args: EventTypes` means `args: tuple[EventTypes, ...]`. But I understand that you have left `list` for backward compatibility.
